plugin/advisor/advisor.go

```diff
@@ -188,3 +195,21 @@ func Check(dbType db.Type, advType Type, ctx Context, statement string) ([]Advic
 
 	return f.Check(ctx, statement)
 }
+
+// IsSyntaxCheckSupported checks the engine type if syntax check supports it.
+func IsSyntaxCheckSupported(engine db.Type) bool {
+	switch engine {
+	case db.MySQL, db.TiDB, db.Postgres:
+		return true
+	}
+	return false
+}
+
+// IsSchemaReviewSupported checks the engine type if schema review supports it.
+func IsSchemaReviewSupported(engine db.Type) bool {
+	switch engine {
+	case db.MySQL, db.TiDB:
```

Comment:
And Postgres? I guess we can use if instead of switch for this case?

Reply:
Aha, `if` is the correct way.

But this PR only supports the PostgreSQL syntax check, I'll add `Postgres` for schema review supported check in the following PR.